src/transforms/repair.jl

```diff
@@ -0,0 +1,61 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Repair{K}
+
+Perform repairing operation with code `K`.
+
+## Available operations
+
+- K = 0: duplicated vertices and faces are removed
+- K = 1: unused vertices are removed
+- K = 2: non-manifold faces are removed
+- K = 3: degenerate faces are removed
+- K = 4: non-manifold vertices are removed
+- K = 5: non-manifold vertices are split by threshold
+- K = 6: close vertices are merged (given a radius)
+- K = 7: faces are coherently oriented
+"""
+struct Repair{K} <: StatelessGeometricTransform end
+
+# implement operation K = 1
+function apply(transform::Repair{1}, mesh)
+  # get the faces
+  topo = convert(HalfEdgeTopology, topology(mesh))
+  ∂₂₀ = Boundary{2,0}(topo)
+  nelem = nelements(topo)
+
+  # indices of used vertices will be stored in this vector
+  used = Vector{Int}(undef, 0)
+
+  # initialize dictionary to map old indices to new indices
+  newindices = Dict{Int,Int}()
+
+  # to store the connectivities of the cleaned mesh
+  connec = Vector{Connectivity}(undef, 0)
+
+  # iterate over elements
+  nused = 0
+  for e in 1:nelem
+    elem = ∂₂₀(e)
+    for i in elem
+      if i ∉ used
+        push!(used, i)
+        nused += 1
+        newindices[i] = nused
+      end
+    end
+    c = connect(tuple([newindices[i] for i in elem]...))
```

Comment:
what happens when the dictionary doesn't have the index i?

Reply:
also, use ntuple(i->newindices[i], elem) instead to create the tuple statically

Reply:
I believe it's not possible that the dictionary doesn't have the index i (it's late and I can't seriously think about that right now).

Your code `ntuple(i->newindices[i], elem)` does not work.